GitFyle.Core.Api/Services/Foundations/Sources/SourceService.Validations.cs

```diff
@@ -106,16 +131,25 @@ private static async ValueTask<bool> IsExceedingLengthAsync(string text, int max
             Message = "Url is invalid"
         };
 
-        private static async ValueTask<dynamic> IsDatesNotSameAsync(
-            DateTimeOffset createdDate,
-            DateTimeOffset updatedDate,
-            string createdDateName) => new
+        private static async ValueTask<dynamic> IsSameAsync(
+            DateTimeOffset firstDate,
+            DateTimeOffset secondDate,
+            string secondDateName) => new
+            {
+                Condition = firstDate == secondDate,
+                Message = $"Date is the same as {secondDateName}"
+            };
+
+        private static async ValueTask<dynamic> IsNotSameAsync(
+            DateTimeOffset firstDate,
+            DateTimeOffset secondDate,
+            string secondDateName) => new
             {
-                Condition = createdDate != updatedDate,
-                Message = $"Date is not the same as {createdDateName}"
+                Condition = firstDate != secondDate,
+                Message = $"Date is not the same as {secondDateName}"
             };
 
-        private static async ValueTask<dynamic> IsValuesNotSameAsync(
+        private static async ValueTask<dynamic> IsNotSameAsync(
             string createBy,
             string updatedBy,
             string createdByName) => new
```

Comment:
This is a generic method that could be used on any string.

```suggestion
        private static async ValueTask<dynamic> IsNotSameAsync(
            string first,
            string second,
            string secondName) => new
```